contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1625 @@
+                    _beneficiaryAddress,
+                    _toucanPool,
+                    redeemAmount[0]
+                );
+                emit ToucanRetired(
+                    msg.sender,
+                    _beneficiaryAddress,
+                    _beneficiaryString,
+                    _retirementMessage,
+                    _toucanPool,
+                    _listTCO2[i],
+                    redeemAmount[0]
+                );
+
+                leftToBurn -= redeemAmount[0];
+            }
+        }
+
+        require(
+            leftToBurn == 0,
+            "Not all pool tokens were burned. Please submit a larger TCO2 token list."
+        );
+    }
+
+    /**
+        @notice Unwraps/unstakes any KLIMA needed to regular KLIMA.
+        @param _klimaType Address of the KLIMA type being used.
+        @param _amountIn Amount of pool token needed.
+     */
+    function _stakedToUnstaked(address _klimaType, uint256 _amountIn)
+        internal
+        returns (uint256)
+    {
+        uint256 unwrappedKLIMA;
+
+        if (_klimaType == wsKLIMA) {
+            // Get wsKLIMA needed, transfer and unwrap, unstake to KLIMA
+            uint256 wsKLIMANeeded = IwsKLIMA(wsKLIMA).sKLIMATowKLIMA(_amountIn);
+
+            IERC20(wsKLIMA).safeTransferFrom(
+                msg.sender,
+                address(this),
+                wsKLIMANeeded
+            );
+            IERC20(wsKLIMA).approve(wsKLIMA, wsKLIMANeeded);
+            unwrappedKLIMA = IwsKLIMA(wsKLIMA).unwrap(wsKLIMANeeded);
+            IERC20(sKLIMA).safeIncreaseAllowance(staking, unwrappedKLIMA);
+            IStaking(staking).unstake(unwrappedKLIMA, false);
+        }
+
+        // If using sKLIMA, transfer in and unstake
+        if (_klimaType == sKLIMA) {
+            IERC20(sKLIMA).safeTransferFrom(
+                msg.sender,
+                address(this),
+                _amountIn
+            );
+            IERC20(sKLIMA).safeIncreaseAllowance(staking, _amountIn);
+            IStaking(staking).unstake(_amountIn, false);
+        }
+
+        return unwrappedKLIMA;
+    }
+
+    /**
+        @notice Bonds any accumualted fees to the DAO address if the minimum bond size is met.
+        @param _klimaType Address of the type of KLIMA provided.
+        @param _amounts Swap results.
+        @param _amountIn Original KLIMA amount needed.
+        @param _unwrappedKLIMA Original amount of KLIMA that was unwrapped.
+     */
+    function _returnKLIMADust(
+        address _klimaType,
+        uint256[] memory _amounts,
+        uint256 _amountIn,
+        uint256 _unwrappedKLIMA
+    ) internal {
+        uint256 tradeDust;
+
+        if (_klimaType == KLIMA) {
+            tradeDust =
+                _amountIn -
+                (_amounts[0] == 0 ? _amounts[1] : _amounts[0]);
+            IERC20(KLIMA).safeTransfer(msg.sender, tradeDust);
+        }
+        if (_klimaType == sKLIMA) {
```

Comment:
Maybe change these to `else if` statements to avoid the extra jumps in case type is KLIMA or sKLIMA.